### PR TITLE
Enhanced RDoc for GetoptLong

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Enhanced RDoc for GetoptLong.

### CODE REVIEW

**Comment by hsbt** [comment]
I'm not native. I couldn't review this.

File: lib/getoptlong.rb
Comment (jeremyevans) [inline lib/getoptlong.rb:106]: Seems weird to call this "protected options"?  Does GetoptLong use that wording internally?  `--` on the command line indicates that all following entries in argv are arguments and not options, even if they start with a dash.

File: lib/getoptlong.rb
Comment (jeremyevans) [inline lib/getoptlong.rb:19]: How does this including work for usage as a default gem in Ruby?  I'm guessing it wouldn't work, since we don't want to have `simple.rb` in `doc` in the Ruby repository.  Possibly we could use a `doc/getoptlong` directory instead.  Not sure if that is desired.  Hopefully @hsbt can provide input.

File: lib/getoptlong.rb
Comment (BurdetteLamar) [reply lib/getoptlong.rb:106]: I agree that "protected options" is weak, and is not used internally.  I don't have a better term to use as the heading.

File: lib/getoptlong.rb
Comment (BurdetteLamar) [reply lib/getoptlong.rb:19]: Will wait for direction on this one.

File: lib/getoptlong.rb
Comment (jeremyevans) [reply lib/getoptlong.rb:106]: Maybe "Treating Remaining Options as Arguments"?

File: lib/getoptlong.rb
Comment (BurdetteLamar) [reply lib/getoptlong.rb]: Have moved included files to doc/getoptlong, which is consistent with other default gems' usage.  Also (the horror!), I did not realize that directory doc was in the .gitignore file, and so only now are the .rdoc files getting into this PR.

**Comment by BurdetteLamar** [comment]
I see that there's no maintainer for this repo.  So how does this (or anything) get reviewed and merged?

File: lib/getoptlong.rb
Comment (jeremyevans) [inline lib/getoptlong.rb:201]: I would remove discussion of command lines.  Outside of Windows, programs in general are called with array of arguments, not with a command line.  If you are executing commands from the shell, the shell will take the command line, parse it, and call the program with an array of arguments.  You can focus discussion on ARGV, without discussing command lines.

File: .gitignore
Comment (jeremyevans) [inline .gitignore:5]: `/doc/` may be ignored for a reason.  Maybe because it is the default location for `rdoc` to generate documentation?  I recommend having the example code in an `examples` directory, as that better indicates that these are executable examples.  You can still include the code in the documentation with `:include:`

**Review by jeremyevans — CHANGES_REQUESTED** [review]
I think this is a major improvement to the documentation, I just have a couple of requests for changes.

File: lib/getoptlong.rb
Comment (BurdetteLamar) [reply lib/getoptlong.rb:201]: Do you mean just here?  Or throughout (28 mentions of command line, including example command lines)?

Comment (jeremyevans) [reply lib/getoptlong.rb:201]: When discussing example command lines, it's fine to keep the reference.  However, any place you are discussing argument processing, the reference should be removed.  It's important to understand that the command line to argument processing is unrelated to option parsing.  Option parsing via optparse/getoptlong is for taking an array of arguments and separating out the options.  From a lexing/parsing perspective, the shell lexes, and optparse/getoptlong parses.

File: .gitignore
Comment (BurdetteLamar) [reply .gitignore]: Moved to examples directory.

**Comment by BurdetteLamar** [comment]
@jeremyevans, ready for review.

**Review by jeremyevans — APPROVED** [review]
`examples` change looks fine.  I still think it would be a good idea to change the command line references, but I'll leave that to you.  `getoptlong` is currently unmaintained, so you can merge this yourself after you decide whether to change the references to the command line.

**Comment by BurdetteLamar** [comment]
> `examples` change looks fine. I still think it would be a good idea to change the command line references, but I'll leave that to you. `getoptlong` is currently unmaintained, so you can merge this yourself after you decide whether to change the references to the command line.

@jeremyevans, I'm happy to make these further changes.  I've removed a dozen referenced to the command line, leaving only those that refer to an actual example command line.  Hope I haven't removed too many.

**Pull request merged**
